packages/core/src/codewhispererChat/tools/fsRead.ts

```diff
@@ -0,0 +1,175 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { getLogger } from '../../shared/logger/logger'
+
+// Constants
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const MAX_TOOL_RESPONSE_SIZE = 30720 // 30KB
+
+enum OutputKind {
+    Text = 'text',
+    Json = 'json',
+}
+
+export interface InvokeOutput {
+    output: {
+        kind: OutputKind
+        content: string
+    }
+}
+
+export interface FsReadParams {
+    path: string
+    readRange?: number[]
+}
+
+export class FsRead {
+    private readonly fsPath: string
+    private readonly readRange?: number[]
+
+    constructor(params: FsReadParams) {
+        this.fsPath = params.path
+        this.readRange = params.readRange
+    }
+
+    public async invoke(): Promise<InvokeOutput> {
+        try {
+            const fileUri = vscode.Uri.file(this.fsPath)
+            let fileStat: vscode.FileStat
+            try {
+                fileStat = await vscode.workspace.fs.stat(fileUri)
```

Comment:
you might want to take a look at our [fs](https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/shared/fs/fs.ts) api instead if you ever need this to work in both the browser and node

Reply:
Wait actually can we just switch to our fs api anyways? It should be a quick change. Then if this code ever needs to run in the browser we're all set